### PR TITLE
Operator v1 Migration

### PR DESCRIPTION
**Pull request by adambkaplan**
Migrate operator API to `operator/v1`, along with corresponding changes to `library-go` methods.

Also removed the `public-info` configMap.

### CODE REVIEW

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go [hunk 1]
```diff
@@ -124,11 +157,6 @@ func manageOpenShiftControllerManagerPublicConfigMap_v311_00_to_latest(client co
 	apiserverConfig := uncastUnstructured.(runtime.Unstructured)
 
 	configMap := resourceread.ReadConfigMapV1OrDie(v311_00_assets.MustAsset("v3.11.0/openshift-controller-manager/public-info.yaml"))
-	if operatorConfig.Status.CurrentAvailability != nil {
-		configMap.Data["version"] = operatorConfig.Status.CurrentAvailability.Version
-	} else {
-		configMap.Data["version"] = ""
-	}
```
[the remark this answers is not in the document]
Comment (bparees) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00.go]: i'm not familiar with it.  @deads2k ?

it's worth noting that it was recording internalRegistryHostname before i even introduced the config observer for internalRegistryHostname, if you look at the commit dates.

**Comment by bparees** [comment]
@sanchezl you likely have experience here as well (migrating the api)

**Comment by adambkaplan** [comment]
ping @deads2k @sanchezl

**Comment by bparees** [comment]
@adambkaplan i think we have enough coverage at this point that if it passes we can merge it.

/retest

**Comment by bparees** [comment]
(unless there is anything else you still needed to do?)

**Comment by adambkaplan** [comment]
@bparees I'd like to rip out that public-info ConfigMap, then squash. I believe the internal and external hostname are passed to the controller config YAML/JSON, so we don't need it.

**Comment by bparees** [comment]
> @bparees I'd like to rip out that public-info ConfigMap, then squash. I believe the internal and external hostname are passed to the controller config YAML/JSON, so we don't need it.

do it.

**Comment by adambkaplan** [comment]
Rebased and squashed

**Comment by adambkaplan** [comment]
Looks like e2e tests are blocked until https://github.com/openshift/origin/pull/21614 merges

**Comment by bparees** [comment]
Right but why is this pr causing that to fail and how does the origin pr
fix it?


Ben Parees | OpenShift

**Comment by adambkaplan** [comment]
Error: `certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kube-ca"`. Issue with the image registry using a cert signed by the service-signing controller?

**Comment by bparees** [comment]
We are already populating the service CA in the build pod asdoe from your
origin pr, that's how I demo's this working yesterday. Your pr is doing it
the "right" way, but the logic I put in place a while ago is still there
and should make Buildah trust the registry. It uses the CA that is
automatically mounted into every pod.

So I think your error means something else is wrong.

Ben Parees | OpenShift

**Pull request merged**